django-stubs/db/backends/base/operations.pyi

```diff
@@ -66,16 +69,16 @@ class BaseDatabaseOperations:
     def savepoint_rollback_sql(self, sid: str) -> str: ...
     def set_time_zone_sql(self) -> str: ...
     def sql_flush(
-        self, style: Any, tables: Sequence[str], *, reset_sequences: bool = ..., allow_cascade: bool = ...
+        self, style: Style, tables: Sequence[str], *, reset_sequences: bool = ..., allow_cascade: bool = ...
     ) -> list[str]: ...
-    def execute_sql_flush(self, sql_list: Iterable[str]) -> None: ...
-    def sequence_reset_by_name_sql(self, style: Style | None, sequences: list[Any]) -> list[Any]: ...
-    def sequence_reset_sql(self, style: Style, model_list: Sequence[type[Model]]) -> list[Any]: ...
+    def execute_sql_flush(self, sql_list: list[str]) -> None: ...
+    def sequence_reset_by_name_sql(self, style: Style, sequences: list[dict[str, str | None]]) -> list[str]: ...
+    def sequence_reset_sql(self, style: Style, model_list: list[type[Model]]) -> list[str]: ...
```

Comment:
Since `sequence_reset_sql` only iterates over the `model_list`, I typing this as `list` seems too strict. `Sequence[]` or `Iterable[]` would be better.

Same for `execute_sql_flush`

Reply:
I went for `list` as these methods receive an actual `list` at runtime (e.g. `sequence_reset_sql` receives the result of `sql_flush`), so I assumed it is safe to allow `BaseDatabaseOperations` subclasses to rely on the available attributes/methods of `list` in these methods. On the other hand, keeping `Sequence` seems fine as well